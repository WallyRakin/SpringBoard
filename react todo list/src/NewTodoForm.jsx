// NewTodoForm.js
import React, { useState } from 'react';


export function NewTodoForm({ addTodo }) {
    const [task, setTask] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();
        addTodo({ task });
        setTask('');
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={task}
                onChange={(e) => setTask(e.target.value)}
                placeholder="Enter a new task"
            />
            <button type="submit">Add Todo</button>
        </form>
    );
};