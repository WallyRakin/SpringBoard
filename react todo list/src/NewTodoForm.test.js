import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { NewTodoForm } from './NewTodoForm';

it('NewTodoForm renders without crashing', () => {
    render(<NewTodoForm addTodo={() => { }} />);
});

it('NewTodoForm snapshot', () => {
    const tree = renderer.create(<NewTodoForm addTodo={() => { }} />).toJSON();
    expect(tree).toMatchSnapshot();
});
