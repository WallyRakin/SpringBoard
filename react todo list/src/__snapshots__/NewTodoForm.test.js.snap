// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Enter a new task"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Add Todo
  </button>
</form>
`;
